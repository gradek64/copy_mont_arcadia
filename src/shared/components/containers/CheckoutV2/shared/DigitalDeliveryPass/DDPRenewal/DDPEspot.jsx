import React from 'react'
import PropTypes from 'prop-types'

import Espot from '../../../../Espot/Espot'

const DDPEspot = ({ hasDDPExpired, getDDPRenewalEspots }) => {
  if (process.browser) {
    getDDPRenewalEspots()
  }
  const identifier = hasDDPExpired
    ? 'ddp_renewal_expired'
    : 'ddp_renewal_expiring'

  return (
    <div data-espot={identifier}>
      <Espot identifier={identifier} isResponsive />
    </div>
  )
}

DDPEspot.propTypes = {
  getDDPRenewalEspots: PropTypes.func.isRequired,
  hasDDPExpired: PropTypes.bool,
}

DDPEspot.defaultProps = {
  hasDDPExpired: false,
}

export default DDPEspot
