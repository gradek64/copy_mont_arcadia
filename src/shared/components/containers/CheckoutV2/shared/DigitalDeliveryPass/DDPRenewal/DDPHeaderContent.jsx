import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const DDPHeaderContent = ({ isContentCentered, headerText, DDPIcon }) => {
  const titleContainerClass = classNames('DDPRenewal-titleContainer', {
    'DDPRenewal-titleContainer--centered': isContentCentered,
  })

  return (
    <div className={titleContainerClass}>
      {DDPIcon}
      <h3 className="DDPRenewal-title">{headerText}</h3>
    </div>
  )
}

DDPHeaderContent.propType = {
  headerText: PropTypes.string.isRequired,
  DDPIcon: PropTypes.element.isRequired,
  isContentCentered: PropTypes.bool,
}

DDPHeaderContent.defaultProps = {
  isContentCentered: false,
}

export default DDPHeaderContent
