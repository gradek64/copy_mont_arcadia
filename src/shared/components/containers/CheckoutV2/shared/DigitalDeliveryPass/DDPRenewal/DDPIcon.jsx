import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import Image from '../../../../../common/Image/Image'

const DDPIcon = ({ ddpAdded, ddpLogoSrc }) => {
  const className = classnames({
    'DDPAdded-image': ddpAdded,
    'DDPRenewal-image': !ddpAdded,
  })
  return <Image className={className} src={ddpLogoSrc} />
}

DDPIcon.propTypes = {
  ddpLogoSrc: PropTypes.string.isRequired,
  ddpAdded: PropTypes.bool,
}

DDPIcon.defaultProps = {
  ddpAdded: false,
}

export default DDPIcon
