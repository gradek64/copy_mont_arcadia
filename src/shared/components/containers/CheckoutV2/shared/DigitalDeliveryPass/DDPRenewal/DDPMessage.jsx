import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const DDPMessage = ({ isContentCentered, message, children }) => {
  const messageClass = classNames('DDPRenewal-message', {
    'DDPRenewal-message--centered': isContentCentered,
  })
  return (
    <span className={messageClass}>
      {message ? (
        // eslint-disable-next-line react/no-danger
        <span dangerouslySetInnerHTML={{ __html: message }} />
      ) : (
        children || ''
      )}
    </span>
  )
}

DDPMessage.propTypes = {
  children: PropTypes.element,
  message: PropTypes.string,
  isContentCentered: PropTypes.bool,
}

DDPMessage.defaultProps = {
  isContentCentered: false,
}

export default DDPMessage
