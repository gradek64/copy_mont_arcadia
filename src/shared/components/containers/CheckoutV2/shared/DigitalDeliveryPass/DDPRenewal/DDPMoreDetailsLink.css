/** @define DDPMoreDetailsLink */

.DDPMoreDetailsLink{
  margin: 20px 0 30px 0;
}
