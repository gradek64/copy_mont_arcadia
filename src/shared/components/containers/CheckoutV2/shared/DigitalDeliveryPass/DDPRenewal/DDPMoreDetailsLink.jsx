import React from 'react'
import { Link } from 'react-router'
import PropTypes from 'prop-types'

const landingPagesForDDP = {
  tsuk: '/en/tsuk/category/topshop-premier-8994768/home',
  tmuk: '/en/tmuk/category/topman-premier-8995089/home',
  dpuk: '/en/dpuk/category/digital-delivery-pass-8069175/home',
  msuk: '/en/msuk/category/miss-s-unlimited-8073836/home',
  bruk: '/en/bruk/category/ddp-8068633/home',
  wluk: '/en/wluk/category/unlimited-delivery-8036879/home',
  evuk: '/en/evuk/category/unlimited-delivery-8060594/home?TS=1566400554634',
}

const DDPMoreDetailsLink = ({ storeCode }) => {
  if (!landingPagesForDDP[storeCode]) return null
  return (
    <Link
      className="DDPMoreDetailsLink"
      to={landingPagesForDDP[storeCode]}
      target="_blank"
    >
      More details
    </Link>
  )
}

DDPMoreDetailsLink.propTypes = {
  storeCode: PropTypes.string.isRequired,
}

export default DDPMoreDetailsLink
