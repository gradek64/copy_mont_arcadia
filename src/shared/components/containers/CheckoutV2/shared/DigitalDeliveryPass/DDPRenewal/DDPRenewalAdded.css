/** @define DDPAdded */

.DDPAdded {
  padding: $ddp-container-padding;
  align-items: $ddp-added-align-items;
  background: $ddp-added-title-background-colour;
  border: $ddp-added-border;
  display: $ddp-added-display;
  height: $ddp-added-height;
  margin-bottom: 30px;

  &-outer--mydetails {
    border-top: 1px solid #979797;
    padding: 10px;
  }

  &-container {
    display: $ddp-added-container-display;
    min-height: $ddp-added-container-height;
  }

  &-title {
    padding: $ddp-added-title-padding;
    text-transform: $ddp-added-title-text-transform;
    font-weight: $ddp-added-title-font-weight;
    color: $ddp-added-title-color;
  }

  &-withOuterDDPTitleAndSeparatorTitle {
    margin-top: 37px;
    margin-bottom: 30px;
    font-size: 17px;
    font-weight: $ddp-renewal-font-weight-title;
  }

  &-titleContainer {
    display: $ddp-added-title-conatiner-display;
    align-items: $ddp-added-title-container-align-items;
  }

  &-message {
    margin-bottom: $ddp-added-message-margin-bottom;
    text-align: $ddp-added-message-text-align;
    font-weight: $ddp-added-message-font-weight;
    color: $ddp-added-message-color;
  }

  &-icon {
    padding-right: 10px;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &-image {
    width: $ddp-added-icon-width;
    height: $ddp-added-icon-height;
    fill: $ddp-added-icon-fill;
  }

  &--myaccount,
  &--mydetails {
    padding-top: 10px;
    padding-bottom: 18px;
    height: auto;
    flex-direction: column;
    justify-content: center;
    text-align: center;

    & .DDPAdded-title {
      margin-top: 0;
      margin-bottom: 5px;
      padding: 0;
    }

    & .DDPAdded-container {
      position: relative;
      top: -20px;
    }
    & .DDPAdded-titleContainer {
      justify-content: center;
    }
  }
}

@media $mobile {
  .DDPAdded {
    padding: $ddp-container-padding-mobile;
    height: $ddp-added-height-mobile;
    min-height: $ddp-added-min-height-mobile;

    &-title {
      padding: $ddp-added-title-padding-mobile;
      margin: $ddp-added-title-margin-mobile;
    }

    &-icon {
      padding-right: 0px;
    }

    &-outer--mydetails {
      border-top: 1px solid #979797;
      padding: 10px;
    }

    &-image {
      height: $ddp-added-icon-height-mobile;
      width: $ddp-added-icon-width-mobile;
      align-self: $ddp-added-icon-align-self-mobile;
    }
    
    &--myaccount {
      border: none;
      border-bottom: 1px solid #ccc;
    }

    &--myaccount,
    &--mydetails {
      padding: 10px 40px;
      margin-bottom: 0px;
      
      & .DDPAdded-title {
        font-size: 14px;
      }
      
      & .DDPAdded-image {
        height: $ddp-added-icon-height;
        width: $ddp-added-icon-width;
      }
    }

    &--mydetails {
      padding: 10px 30px 30px 30px;
    }
  }
}
