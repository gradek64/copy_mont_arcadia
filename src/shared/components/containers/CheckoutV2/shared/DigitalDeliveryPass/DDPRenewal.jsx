import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import {
  getDDPDefaultName,
  ddpEndDate,
  isRenewingDDP,
  showDDPRenewal,
  showDDPSavings,
  hasDDPExpired,
  ddpSavingsValue,
  ddpExpressDeliveryPrice,
  ddpLogoSrc,
  ddpAddedLogoSrc,
  showActiveDDP,
  ddpActiveLogoSrc,
  showDDPRenewalWithinDefaultExpiringBoundaries,
} from '../../../../../selectors/ddpSelectors'
import {
  getDDPDefaultSku,
  getDDPDefaultSkuPrice,
} from '../../../../../selectors/siteOptionsSelectors'
import { getStoreCode } from '../../../../../selectors/configSelectors'

import DDPRenewalAdded from '../DigitalDeliveryPass/DDPRenewal/DDPRenewalAdded'
import DDPRenewalExpiry from '../DigitalDeliveryPass/DDPRenewalExpiry'
import DDPIcon from '../DigitalDeliveryPass/DDPRenewal/DDPIcon'
import DDPRenewalActive from '../DigitalDeliveryPass/DDPRenewalActive'

// Actions
import { addDDPToBag } from '../../../../../actions/common/ddpActions'
import {
  getDDPRenewalEspots,
  getDDPTermsAndConditions,
} from '../../../../../actions/common/espotActions'
import { showTAndCModal } from '../../../../../actions/common/modalActions'

const DDPRenewal = (
  {
    // props
    showDDPRenewal,
    isRenewingDDP,
    ddpProductName,
    getDDPDefaultSku,
    ddpEndDate,
    showDDPSavings,
    hasDDPExpired,
    ddpSavingsValue,
    ddpPrice,
    ddpExpressDeliveryPrice,
    ddpLogoSrc,
    ddpAddedLogoSrc,
    ddpActiveLogoSrc,
    withMoreDetailsLink,
    storeCode,
    withOuterDDPTitleAndSeparator,
    showActiveDDP,
    isMyAccount,
    isMyAccountDetail,
    showDDPRenewalWithinDefaultExpiringBoundaries,
    // Actions
    addToBag,
    getDDPRenewalEspots,
    showTAndCModal,
    getDDPTermsAndConditions,
  },
  { l, p }
) => {
  getDDPTermsAndConditions()

  const canShowActiveDDP = !isRenewingDDP && showActiveDDP && isMyAccountDetail

  const renderDDPRenewalExpiry = isMyAccount
    ? showDDPRenewalWithinDefaultExpiringBoundaries // Shows the DDPRenewalExpiry component 30 days prior and 30 days after expiry.
    : showDDPRenewal // Shows the DDPRenewalExpiry component 30 days prior and indefinitely after expiry.

  if (renderDDPRenewalExpiry) {
    return (
      <DDPRenewalExpiry
        isMyAccount={isMyAccount}
        isMyAccountDetail={isMyAccountDetail}
        ddpSku={getDDPDefaultSku.sku}
        ddpProductName={ddpProductName}
        ddpEndDate={ddpEndDate}
        showDDPSavings={showDDPSavings}
        hasDDPExpired={hasDDPExpired}
        ddpSavingsValue={p(ddpSavingsValue)}
        ddpFormattedPrice={p(ddpPrice)}
        ddpExpressDeliveryPrice={p(ddpExpressDeliveryPrice)}
        addToBag={addToBag}
        getDDPRenewalEspots={getDDPRenewalEspots}
        DDPIcon={<DDPIcon ddpLogoSrc={ddpLogoSrc} />}
        withMoreDetailsLink={withMoreDetailsLink}
        storeCode={storeCode}
        withOuterDDPTitleAndSeparator={withOuterDDPTitleAndSeparator}
        openModal={showTAndCModal}
        l={l}
      />
    )
  } else if (isRenewingDDP) {
    return (
      <DDPRenewalAdded
        isMyAccount={isMyAccount}
        isMyAccountDetail={isMyAccountDetail}
        productName={ddpProductName}
        DDPIcon={<DDPIcon ddpAdded ddpLogoSrc={ddpAddedLogoSrc} />}
        ddpEndDate={ddpEndDate}
        l={l}
      />
    )
  } else if (canShowActiveDDP) {
    return (
      <DDPRenewalActive
        ddpSku={getDDPDefaultSku.sku}
        ddpProductName={ddpProductName}
        ddpEndDate={ddpEndDate}
        showDDPSavings={showDDPSavings}
        hasDDPExpired={hasDDPExpired}
        ddpSavingsValue={p(ddpSavingsValue)}
        ddpFormattedPrice={p(ddpPrice)}
        ddpExpressDeliveryPrice={p(ddpExpressDeliveryPrice)}
        addToBag={addToBag}
        getDDPRenewalEspots={getDDPRenewalEspots}
        DDPIcon={<DDPIcon ddpLogoSrc={ddpActiveLogoSrc} />}
        isContentCentered={isMyAccountDetail}
        storeCode={storeCode}
        withOuterDDPTitleAndSeparator={withOuterDDPTitleAndSeparator}
        isMyAccount={isMyAccount}
        isMyAccountDetail={isMyAccountDetail}
        l={l}
      />
    )
  }
  return null
}

DDPRenewal.propTypes = {
  // Properties
  showDDPRenewal: PropTypes.bool.isRequired,
  isRenewingDDP: PropTypes.bool.isRequired,
  ddpProductName: PropTypes.string.isRequired,
  getDDPDefaultSku: PropTypes.shape({
    sku: PropTypes.string.isRequired,
    default: PropTypes.bool.isRequired,
    unitPrice: PropTypes.number.isRequired,
    catentryId: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    timePeriod: PropTypes.string.isRequired,
  }).isRequired,
  ddpEndDate: PropTypes.string.isRequired,
  showDDPSavings: PropTypes.bool.isRequired,
  hasDDPExpired: PropTypes.bool.isRequired,
  ddpSavingsValue: PropTypes.number.isRequired,
  ddpPrice: PropTypes.number.isRequired,
  ddpExpressDeliveryPrice: PropTypes.number.isRequired,
  storeCode: PropTypes.string.isRequired,
  withMoreDetailsLink: PropTypes.bool,
  withOuterDDPTitleAndSeparator: PropTypes.bool,
  ddpLogoSrc: PropTypes.string.isRequired,
  ddpAddedLogoSrc: PropTypes.string.isRequired,
  ddpActiveLogoSrc: PropTypes.string.isRequired,
  showDDPRenewalWithinDefaultExpiringBoundaries: PropTypes.bool.isRequired,
  isMyAccount: PropTypes.bool,
  isMyAccountDetail: PropTypes.bool,

  // Actions
  addToBag: PropTypes.func.isRequired,
  getDDPRenewalEspots: PropTypes.func.isRequired,
  showTAndCModal: PropTypes.func.isRequired,
  getDDPTermsAndConditions: PropTypes.func.isRequired,
}

DDPRenewal.defaultProps = {
  ddpEndDate: '', // if DDP has expired, no end date is provided
  withMoreDetailsLink: false,
  withOuterDDPTitleAndSeparator: false,
  isMyAccount: false,
  isMyAccountDetail: false,
}

DDPRenewal.contextTypes = {
  l: PropTypes.func.isRequired,
  p: PropTypes.func.isRequired,
}

const mapStateToProps = (state) => ({
  showDDPRenewal: showDDPRenewal(state),
  isRenewingDDP: isRenewingDDP(state),
  getDDPDefaultSku: getDDPDefaultSku(state),
  ddpEndDate: ddpEndDate(state),
  showDDPSavings: showDDPSavings(state),
  hasDDPExpired: hasDDPExpired(state),
  ddpSavingsValue: ddpSavingsValue(state),
  ddpPrice: getDDPDefaultSkuPrice(state),
  ddpExpressDeliveryPrice: ddpExpressDeliveryPrice(state),
  ddpLogoSrc: ddpLogoSrc(state),
  ddpAddedLogoSrc: ddpAddedLogoSrc(state),
  ddpActiveLogoSrc: ddpActiveLogoSrc(state),
  ddpProductName: getDDPDefaultName(state),
  storeCode: getStoreCode(state),
  showActiveDDP: showActiveDDP(state),
  showDDPRenewalWithinDefaultExpiringBoundaries: showDDPRenewalWithinDefaultExpiringBoundaries(
    state
  ),
})

const mapDispatchToProps = {
  addToBag: addDDPToBag,
  getDDPRenewalEspots,
  showTAndCModal,
  getDDPTermsAndConditions,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DDPRenewal)
