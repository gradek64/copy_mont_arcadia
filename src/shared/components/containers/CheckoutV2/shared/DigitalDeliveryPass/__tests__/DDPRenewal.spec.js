import { compose } from 'ramda'
import {
  mountRender,
  withStore,
  buildComponentRender,
} from 'test/unit/helpers/test-component'

import DDPRenewal from '../DDPRenewal'

const testAction = {
  type: 'TEST',
  payload: 'test',
}

jest.mock('../../../../../../actions/common/espotActions', () => ({
  getDDPRenewalEspots: jest.fn(() => {}),
  getDDPTermsAndConditions: jest.fn(() => ({
    type: 'TEST',
    payload: 'test',
  })),
}))

describe('<DDPRenewal /> Mounted', () => {
  const testExpiredUrl = 'test-expired/Url'
  const testExpiringUrl = 'test-expiring/Url'

  const testTermsUrl = 'test-terms/Url'
  const createInitialState = (user, products = []) => ({
    account: {
      user: {
        isDDPUser: true,
        isDDPRenewable: true,
        ddpStartDate: '16 May 2019',
        ddpEndDate: '15 May 2020',
        wasDDPUser: false,
        ddpCurrentOrderCount: 2,
        ddpPreviousOrderCount: null,
        ddpCurrentSaving: 2.05,
        ddpPreviousSaving: null,
        ddpStandardPrice: 9.95,
        ddpExpressDeliveryPrice: 6.0,
        ...user,
      },
    },
    shoppingBag: {
      bag: { products },
    },
    config: {
      brandDisplayName: 'topshop',
      brandName: 'topshop',
      brandCode: 'ts',
      storeCode: 'tsuk',
    },
    viewport: {
      media: 'mobile',
    },
    routing: {
      visited: [],
    },
    features: {
      status: {
        FEATURE_DDP: true,
        FEATURE_IS_DDP_RENEWABLE: true,
      },
    },
    sandbox: {
      pages: {},
    },
    siteOptions: {
      ddp: {
        ddpProduct: {
          ddpSkus: [
            {
              sku: '100000012',
              wasPrice: 7,
              default: true,
              unitPrice: 4,
              catentryId: '32482151',
              wasWasPrice: 10,
              name: 'DDP VIP Subscription',
              description: 'DDP VIP 12 Months',
              timePeriod: '12',
            },
          ],
          name: 'DDP VIP Subscription',
        },
      },
    },
    espot: {
      cmsData: {
        ddp_renewal_expired: {
          responsiveCMSUrl: testExpiredUrl,
        },
        ddp_renewal_expiring: {
          responsiveCMSUrl: testExpiringUrl,
        },
        ddp_terms_and_conditions: {
          responsiveCMSUrl: testTermsUrl,
        },
      },
    },
  })

  const render = compose(
    mountRender,
    withStore(createInitialState())
  )
  const renderRenewalComponent = buildComponentRender(render, DDPRenewal)

  it('renders', () => {
    const { getTree } = renderRenewalComponent()
    expect(getTree()).toMatchSnapshot()
  })

  describe('<DDPRenewalExpiry />', () => {
    describe('when `isMyAccount` is not equal to true', () => {
      it('should render when showDDPRenewal is true', () => {
        const userState = {
          isDDPRenewable: false,
          isDDPUser: false,
          wasDDPUser: true,
        }
        const render = compose(
          mountRender,
          withStore(createInitialState(userState))
        )
        const renderRenewalComponent = buildComponentRender(render, DDPRenewal)
        const { wrapper } = renderRenewalComponent()

        const ddpRenewal = wrapper.find('.DDPRenewal')
        expect(ddpRenewal).toHaveLength(1)

        const ddpAdded = wrapper.find('.DDPAdded')
        expect(ddpAdded).toHaveLength(0)

        const ddpRenewalActive = wrapper.find('.DDPRenewal-active')
        expect(ddpRenewalActive).toHaveLength(0)
      })

      it('should not render when showDDPRenewal is false', () => {
        const userState = {
          isDDPRenewable: false,
          isDDPUser: false,
          wasDDPUser: false,
        }
        const render = compose(
          mountRender,
          withStore(createInitialState(userState))
        )
        const renderRenewalComponent = buildComponentRender(render, DDPRenewal)
        const { wrapper } = renderRenewalComponent()

        const ddpRenewal = wrapper.find('.DDPRenewal')
        expect(ddpRenewal).toHaveLength(0)
      })
    })

    describe('when `isMyAccount` is equal to true', () => {
      it('should render when showDDPRenewalWithinDefaultExpiringBoundaries is true', () => {
        const { wrapper } = renderRenewalComponent({ isMyAccount: true })
        const ddpRenewal = wrapper.find('.DDPRenewal')
        expect(ddpRenewal).toHaveLength(1)

        const ddpAdded = wrapper.find('.DDPAdded')
        expect(ddpAdded).toHaveLength(0)

        const ddpRenewalActive = wrapper.find('.DDPRenewal-active')
        expect(ddpRenewalActive).toHaveLength(0)
      })

      it('should not render when showDDPRenewalWithinDefaultExpiringBoundaries is false', () => {
        const userState = {
          isDDPRenewable: false,
        }
        const render = compose(
          mountRender,
          withStore(createInitialState(userState))
        )
        const renderRenewalComponent = buildComponentRender(render, DDPRenewal)
        const { wrapper } = renderRenewalComponent({ isMyAccount: true })
        const ddpRenewal = wrapper.find('.DDPRenewal')
        expect(ddpRenewal).toHaveLength(0)
      })
    })
  })

  describe('<DDPRenewalAdded />', () => {
    it('renders when shopping bag contains DDP product and canShowAdded is true', () => {
      const userState = {}
      const products = [{ isDDPProduct: true }]
      const render = compose(
        mountRender,
        withStore(createInitialState(userState, products))
      )
      const renderAddedComponent = buildComponentRender(render, DDPRenewal)
      const { wrapper } = renderAddedComponent({ canShowAdded: true })
      const ddpRenewal = wrapper.find('.DDPRenewal')
      expect(ddpRenewal).toHaveLength(0)

      const ddpRenewalActive = wrapper.find('.DDPRenewal-active')
      expect(ddpRenewalActive).toHaveLength(0)

      const ddpAdded = wrapper.find('.DDPAdded')
      expect(ddpAdded).toHaveLength(1)
    })
  })

  describe('<DDPRenewalActive />', () => {
    it('renders when showActiveDDP, canShowActive are true and shopping bag does not contain DDP product', () => {
      const userState = { isDDPRenewable: false }
      const products = []
      const render = compose(
        mountRender,
        withStore(createInitialState(userState, products))
      )
      const renderAddedComponent = buildComponentRender(render, DDPRenewal)
      const { wrapper } = renderAddedComponent({
        showActiveDDP: true,
        isMyAccountDetail: true,
      })

      const ddpAdded = wrapper.find('.DDPAdded')
      expect(ddpAdded).toHaveLength(0)

      const ddpRenewalActive = wrapper.find('.DDPRenewal.DDPRenewal-active')
      expect(ddpRenewalActive).toHaveLength(1)
    })
  })

  it('should format prices properly', () => {
    const userState = {
      ddpCurrentSaving: 15.2,
      ddpExpressDeliveryPrice: 6.0,
    }
    const render = compose(
      mountRender,
      withStore(createInitialState(userState))
    )
    const renderAddedComponent = buildComponentRender(render, DDPRenewal)
    const { wrapper } = renderAddedComponent()
    const ddpRenewalExpiryWrapper = wrapper.find('DDPRenewalExpiry')
    expect(ddpRenewalExpiryWrapper.prop('ddpSavingsValue')).toBe('£15.20')
    expect(ddpRenewalExpiryWrapper.prop('ddpFormattedPrice')).toBe('£4.00')
    expect(ddpRenewalExpiryWrapper.prop('ddpExpressDeliveryPrice')).toBe(
      '£6.00'
    )
  })

  it('should call getDDPTermsAndConditions', () => {
    const userState = {
      ddpCurrentSaving: 15.2,
      ddpExpressDeliveryPrice: 6.0,
    }
    const render = compose(
      mountRender,
      withStore(createInitialState(userState))
    )
    const renderAddedComponent = buildComponentRender(render, DDPRenewal)
    const { store } = renderAddedComponent()
    expect(store.getActions()[0]).toEqual(testAction)
  })
})
