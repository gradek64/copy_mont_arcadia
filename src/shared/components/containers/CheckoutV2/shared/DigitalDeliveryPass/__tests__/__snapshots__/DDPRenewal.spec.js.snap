// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DDPRenewal /> Mounted renders 1`] = `
<Connect(DDPRenewal)>
  <DDPRenewal
    addToBag={[Function]}
    ddpActiveLogoSrc="/assets/topshop/images/ddp-icon.svg"
    ddpAddedLogoSrc="/assets/topshop/images/ddp-added.gif"
    ddpEndDate="15 May 2020"
    ddpExpressDeliveryPrice={6}
    ddpLogoSrc="/assets/topshop/images/ddp-icon.svg"
    ddpPrice={4}
    ddpProductName="DDP VIP Subscription"
    ddpSavingsValue={2.05}
    getDDPDefaultSku={
      Object {
        "catentryId": "32482151",
        "default": true,
        "description": "DDP VIP 12 Months",
        "name": "DDP VIP Subscription",
        "sku": "100000012",
        "timePeriod": "12",
        "unitPrice": 4,
        "wasPrice": 7,
        "wasWasPrice": 10,
      }
    }
    getDDPRenewalEspots={[Function]}
    getDDPTermsAndConditions={[Function]}
    hasDDPExpired={false}
    isMyAccount={false}
    isMyAccountDetail={false}
    isRenewingDDP={false}
    showActiveDDP={true}
    showDDPRenewal={true}
    showDDPRenewalWithinDefaultExpiringBoundaries={true}
    showDDPSavings={false}
    showTAndCModal={[Function]}
    storeCode="tsuk"
    withMoreDetailsLink={false}
    withOuterDDPTitleAndSeparator={false}
  >
    <DDPRenewalExpiry
      DDPIcon={
        <DDPIcon
          ddpAdded={false}
          ddpLogoSrc="/assets/topshop/images/ddp-icon.svg"
        />
      }
      addToBag={[Function]}
      ddpEndDate="15 May 2020"
      ddpExpressDeliveryPrice="£6.00"
      ddpFormattedPrice="£4.00"
      ddpProductName="DDP VIP Subscription"
      ddpSavingsValue="£2.05"
      ddpSku="100000012"
      getDDPRenewalEspots={[Function]}
      hasDDPExpired={false}
      isMyAccount={false}
      isMyAccountDetail={false}
      l={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                "DDP expiring title ",
                " ",
                "",
              ],
              "DDP VIP Subscription",
              "15 May 2020",
            ],
            Array [
              Array [
                "DDP Extend Msg ",
                "",
              ],
              "15 May 2021",
            ],
            Array [
              Array [
                "DDP Button Extend ",
                "",
              ],
              "£4.00",
            ],
            Array [
              Array [
                "DDP Express Saving ",
                "",
              ],
              "£6.00",
            ],
            Array [
              Array [
                "Terms & Conditions",
              ],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "DDP expiring title DDP VIP Subscription 15 May 2020",
            },
            Object {
              "type": "return",
              "value": "DDP Extend Msg 15 May 2021",
            },
            Object {
              "type": "return",
              "value": "DDP Button Extend £4.00",
            },
            Object {
              "type": "return",
              "value": "DDP Express Saving £6.00",
            },
            Object {
              "type": "return",
              "value": "Terms & Conditions",
            },
          ],
        }
      }
      openModal={[Function]}
      showDDPSavings={false}
      storeCode="tsuk"
      withMoreDetailsLink={false}
      withOuterDDPTitleAndSeparator={false}
    >
      <div
        className="DDPRenewal"
      >
        <div
          className="DDPRenewal-container"
        >
          <DDPHeaderContent
            DDPIcon={
              <DDPIcon
                ddpAdded={false}
                ddpLogoSrc="/assets/topshop/images/ddp-icon.svg"
              />
            }
            headerText="DDP expiring title DDP VIP Subscription 15 May 2020"
            isContentCentered={false}
          >
            <div
              className="DDPRenewal-titleContainer"
            >
              <DDPIcon
                ddpAdded={false}
                ddpLogoSrc="/assets/topshop/images/ddp-icon.svg"
              >
                <Connect(Image)
                  className="DDPRenewal-image"
                  src="/assets/topshop/images/ddp-icon.svg"
                >
                  <Image
                    brandCode="ts"
                    brandName="topshop"
                    className="DDPRenewal-image"
                    dispatch={[Function]}
                    imageRef={[Function]}
                    lazyLoad={false}
                    onError={[Function]}
                    onLoad={[Function]}
                    src="/assets/topshop/images/ddp-icon.svg"
                  >
                    <img
                      className="DDPRenewal-image"
                      onError={[Function]}
                      onLoad={[Function]}
                      src="/assets/topshop/images/ddp-icon.svg"
                    />
                  </Image>
                </Connect(Image)>
              </DDPIcon>
              <h3
                className="DDPRenewal-title"
              >
                DDP expiring title DDP VIP Subscription 15 May 2020
              </h3>
            </div>
          </DDPHeaderContent>
          <div
            className="DDPRenewal-bodyContainer"
          >
            <div
              className="DDPRenewal-messageContainer"
            >
              <DDPMessage
                isContentCentered={false}
                message="DDP Express Saving £6.00"
              >
                <span
                  className="DDPRenewal-message"
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "DDP Express Saving £6.00",
                      }
                    }
                  />
                </span>
              </DDPMessage>
              <DDPMessage
                isContentCentered={false}
                message="DDP Extend Msg 15 May 2021"
              >
                <span
                  className="DDPRenewal-message"
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "DDP Extend Msg 15 May 2021",
                      }
                    }
                  />
                </span>
              </DDPMessage>
              <DDPMessage
                isContentCentered={false}
              >
                <span
                  className="DDPRenewal-message"
                >
                  <DDPEspot
                    getDDPRenewalEspots={[Function]}
                    hasDDPExpired={false}
                  >
                    <div
                      data-espot="ddp_renewal_expiring"
                    >
                      <Connect(Espot)
                        identifier="ddp_renewal_expiring"
                        isResponsive={true}
                      >
                        <Espot
                          customClassName=""
                          identifier="ddp_renewal_expiring"
                          isMobile={true}
                          isResponsive={true}
                          qubitid={null}
                          responsiveCMSUrl="test-expiring/Url"
                        >
                          <Connect(SandBox)
                            contentType="espot"
                            isFinalResponsiveEspotSolution={true}
                            location={
                              Object {
                                "pathname": "test-expiring/Url",
                              }
                            }
                            qubitid={null}
                            sandBoxClassName=""
                            shouldGetContentOnFirstLoad={true}
                          >
                            <SandBox
                              cmsPageName={null}
                              contentType="espot"
                              forceMobile={false}
                              getContent={[Function]}
                              getHomePageContent={[Function]}
                              getSegmentedContent={[Function]}
                              hideTacticalMessage={[Function]}
                              isFinalResponsiveEspotSolution={true}
                              isMobile={true}
                              isResponsiveCatHeader={false}
                              lazyLoad={false}
                              location={
                                Object {
                                  "pathname": "test-expiring%2FUrl",
                                  "search": undefined,
                                }
                              }
                              pages={Object {}}
                              qubitid={null}
                              removeContent={[Function]}
                              resetContent={[Function]}
                              sandBoxClassName=""
                              segmentationRequestData={null}
                              setContent={[Function]}
                              setFormDefaultSchema={[Function]}
                              setPageStatusCode={[Function]}
                              shouldGetContentOnFirstLoad={true}
                              showTacticalMessage={[Function]}
                              visited={Array []}
                            />
                          </Connect(SandBox)>
                        </Espot>
                      </Connect(Espot)>
                    </div>
                  </DDPEspot>
                </span>
              </DDPMessage>
              <DDPRenewalCta
                buttonText="DDP Button Extend £4.00"
                clickHandler={[Function]}
                isContentCentered={false}
              >
                <button
                  className="Button DDPRenewal-button"
                  onClick={[Function]}
                >
                  DDP Button Extend £4.00
                </button>
              </DDPRenewalCta>
            </div>
            <span
              className="DDPRenewal-terms"
            >
              <DDPTermsAndConditions
                buttonText="Terms & Conditions"
                openModal={[Function]}
              >
                <div
                  className="DDPTermsAndConditions"
                >
                  <button
                    className="DDPTermsAndConditions-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Terms & Conditions
                  </button>
                </div>
              </DDPTermsAndConditions>
            </span>
          </div>
        </div>
      </div>
    </DDPRenewalExpiry>
  </DDPRenewal>
</Connect(DDPRenewal)>
`;
